src/back/spv/writer.rs

```diff
@@ -807,6 +823,91 @@ impl Writer {
         id
     }
 
+    fn get_type_inner(
```

Comment:
I am totally not satisfied with this, but I couldn't think of any other way to make a function that will get a type inner based on `LookupType`, without modifying the IR. Any comments to refine this is welcome.

Reply:
Why didn't you go with something like `crate::proc::MaybeOwned`? This function signature would be:
```rust
fn get_type_inner<'a>(
  arena: &'a crate::Arena<crate::Type>,
  lookup_ty: LookupType,
) -> MaybeOwned<'a, crate::TypeInner> {..}
```

Reply:
Thank you very much for your commit, it helped me a lot!